Source/Samples/SamplesManager.cpp

```diff
@@ -203,22 +218,17 @@ void SamplesManager::Start()
     context_->AddFactoryReflection<SampleSelectionScreen>();
 
     inspectorNode_ = MakeShared<Scene>(context_);
-    splashScreen_ = MakeShared<SplashScreen>(context_);
     sampleSelectionScreen_ = MakeShared<SampleSelectionScreen>(context_);
-    splashScreen_->SetNextState(sampleSelectionScreen_);
-    splashScreen_->FetchSceneResourcesAsync("Scenes/RenderingShowcase_0.xml");
-    splashScreen_->SetBackgroundImage(context_->GetSubsystem<ResourceCache>()->GetResource<Texture2D>("Textures/StoneDiffuse.dds"));
-    splashScreen_->SetForegroundImage(context_->GetSubsystem<ResourceCache>()->GetResource<Texture2D>("Textures/LogoLarge.png"));
-    splashScreen_->SetProgressImage(context_->GetSubsystem<ResourceCache>()->GetResource<Texture2D>("Textures/TerrainDetail2.dds"));
-    SetState(splashScreen_);
+    SetState(sampleSelectionScreen_);
 
 #if URHO3D_SYSTEMUI
     if (DebugHud* debugHud = context_->GetSubsystem<Engine>()->CreateDebugHud())
         debugHud->ToggleAll();
 #endif
-
+    auto* input = context_->GetSubsystem<Input>();
     SubscribeToEvent(E_RELEASED, [this](StringHash, VariantMap& args) { OnClickSample(args); });
-    SubscribeToEvent(E_KEYUP, [this](StringHash, VariantMap& args) { OnKeyPress(args); });
+    SubscribeToEvent(&sampleSelectionScreen_->dpadAdapter_, E_KEYUP, [this](StringHash, VariantMap& args) { OnArrowKeyPress(args); });
+    SubscribeToEvent(input, E_KEYUP, [this](StringHash, VariantMap& args) { OnKeyPress(args); });
```

Comment:
Do we need this subscription just for CloseSample to work?..

Reply:
Yes